minions/async/src/main/java/io/github/microcks/minion/async/producer/MQTTProducerManager.java

```diff
@@ -103,4 +112,28 @@ public void publishMessage(String topic, String value) {
          me.printStackTrace();
       }
    }
+
+   /**
+    * Get the MQTT topic name corresponding to a AsyncMockDefinition, sanitizing all parameters.
+    */
+   String getTopicName(AsyncMockDefinition definition, EventMessage eventMessage) {
+      logger.infof("AsyncAPI Operation  {} %s", definition.getOperation().getName());
+      // Produce service name part of topic name.
+      String serviceName = definition.getOwnerService().getName().replace(" ", "");
+      serviceName = serviceName.replace("-", "");
+      // Produce version name part of topic name.
+      String versionName = definition.getOwnerService().getVersion().replace(" ", "");
+      // Produce operation name part of topic name.
+      String operationName = definition.getOperation().getName();
+      if (operationName.startsWith("SUBSCRIBE ") || operationName.startsWith("PUBLISH ")) {
+         operationName = operationName.substring(operationName.indexOf(" ") + 1);
+      }
+
+      // replace the parts
+      operationName = ProducerManager.replacePartPlaceholders(eventMessage, operationName);
```

Comment:
Please check if something from `io.github.microcks.util.URIBuilder` can be reused here.

Reply:
as described elsewhere - I'll submit a separate PR